plugins/engines/Engine.h

```diff
@@ -0,0 +1,93 @@
+/* Copyright (c) 2015-2016, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Responsible Author: Cyrille Favreau <cyrille.favreau@epfl.ch>
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef ENGINE_H
+#define ENGINE_H
+
+#include <brayns/common/types.h>
+
+namespace brayns
+{
+
+/**
+ * Abstract implementation of the ray-tracing engine. What we call the
+ * ray-tracing engine is a 3rd party acceleration library, typically OSPRay,
+ * Optix or FireRays, that provides hardware acceleration.
+ * An engine holds a native implementation of a scene, a camera, a frame buffer
+ * and of one or serveral renderers according to the capatilities of the
+ * acceleration library.
+ */
+class Engine
+{
+public:
+
+    virtual ~Engine() {}
+
+    /**
+       Returns the name of the engine
+       @return the name of the engine
+    */
+    virtual std::string name() const = 0;
+
+    /**
+       Commits changes to the engine. This include scene and camera
+       modifications
+    */
+    virtual void commit() = 0;
+
+    /** Renders the current scene and populates the frame buffer accordingly */
+    virtual void render() = 0;
+
+    /** Executes engine specific pre-render operations */
+    virtual void preRender() = 0;
```

Comment:
Hum... The idea of forcing the implementation in the inherited classes is one way to remind the developper that those methods exist, and prevent him/her to put the code in the render() method instead. That's why I'd rather keep them pure, but I'd like to have your opinion on that :-)